llvm/test/CodeGen/Thumb2/mve-vmull-splat.ll

```diff
@@ -190,12 +190,17 @@ entry:
 define arm_aapcs_vfpcc <4 x i64> @sext32_0213_0ext(<8 x i32> %src1, i32 %src2) {
 ; CHECK-LABEL: sext32_0213_0ext:
 ; CHECK:       @ %bb.0: @ %entry
+; CHECK-NEXT:    .vsave {d8, d9}
+; CHECK-NEXT:    vpush {d8, d9}
```

Comment:
The generated code for these tests has gotten longer, was this previously generating invalid code, or is this an unwanted side-effect of this pass?

Reply:
This is something that is due to the new Pass being enabled on Arm. This test, and the other test that has instructions added in this file require the function `handleSubReg` to be called, and such initialise the registers in a different way. This is causing extra instructions on the Arm Architecture, however as a limited number of test cases are effected I deem this an edge case and as such we should allow the compiler to generate the extra instructions.

Reply:
Fair enough, if there's no easy fix then correctness needs to take priority over performance.